### PR TITLE
Loganalyzer enhancements

### PR DESCRIPTION
**Pull request by marian-pritsak**
Update loganalyzer to read stdin
Fixed loganalyzer expect logic

### CODE REVIEW

File: ansible/roles/test/files/tools/loganalyzer/loganalyzer.py [hunk 1]
```diff
@@ -521,10 +548,11 @@ def write_summary_file(run_id, out_dir, analysis_result_per_file):
...
         total_expect_cnt += file_expect_cnt
 
-    out_file.write("-----------------------------\n")
-    out_file.write("TOTAL MATCHES:    %d\n" % total_match_cnt)
-    out_file.write("TOTAL EXPECTED MATCHES:    %d\n" % total_expect_cnt)
-    out_file.write("-----------------------------\n")
+    out_file.write("-----------------------------------\n")
+    out_file.write("TOTAL MATCHES:                  %d\n" % total_match_cnt)
+    out_file.write("TOTAL EXPECTED MATCHES:         %d\n" % total_expect_cnt)
+    out_file.write("TOTAL EXPECTED MISSING MATCHES: %d\n" % len(unused_regex_messages))
```
Comment (qiluo-msft) [inline ansible/roles/test/files/tools/loganalyzer/loganalyzer.py]: >TOTAL EXPECTED MISSING MATCHES:  [](start = 20, length = 32)

Do you need to change back the grep command which modified by Maggie, in this PR?

Comment (maggiemsft) [reply ansible/roles/test/files/tools/loganalyzer/loganalyzer.py]: Yeah, could your please change the loganalyzer_end.yml which process the different result for validation?

Comment (marian-pritsak) [reply ansible/roles/test/files/tools/loganalyzer/loganalyzer.py]: Sure, done

File: ansible/roles/test/files/tools/loganalyzer/loganalyzer.py [hunk 2]
```diff
@@ -96,6 +96,11 @@ def create_start_marker(self):
 
     #---------------------------------------------------------------------
 
+    def is_filename_stdin(self, file_name):
+        return True if file_name == "-" else False
```
Comment (qiluo-msft) [inline ansible/roles/test/files/tools/loganalyzer/loganalyzer.py]: >return True if file_name == "-" else False [](start = 8, length = 42)

return file_name == "-"

File: ansible/roles/test/files/tools/loganalyzer/loganalyzer.py [hunk 3]
```diff
@@ -288,62 +293,65 @@ def analyze_file(self, log_file_path, match_messages_regex, ignore_messages_rege
         #-- indicates whether log analyzer currently is in the log range between start
         #-- and end marker. see analyze_file method.
         in_analysis_range = False
+        stdin_as_input = True if self.is_filename_stdin(log_file_path) else False
```
Comment (qiluo-msft) [inline ansible/roles/test/files/tools/loganalyzer/loganalyzer.py]: >True if self.is_filename_stdin(log_file_path) else False [](start = 25, length = 56)

self.is_filename_stdin(log_file_path)

File: ansible/roles/test/files/tools/loganalyzer/loganalyzer.py (hunk 2 above)
Comment (marian-pritsak) [reply ansible/roles/test/files/tools/loganalyzer/loganalyzer.py]: Done

File: ansible/roles/test/files/tools/loganalyzer/loganalyzer.py (hunk 3 above)
Comment (marian-pritsak) [reply ansible/roles/test/files/tools/loganalyzer/loganalyzer.py]: Done

File: ansible/roles/test/files/tools/loganalyzer/loganalyzer_end.yml [hunk 4]
```diff
@@ -17,11 +17,11 @@
...
 
 - name: Check if loganalyzer missed expected messages
-  shell: grep "TOTAL EXPECTED MATCHES" "{{ test_out_dir }}/{{ summary_file }}" | sed -n "s/TOTAL EXPECTED MATCHES:[[:space:]]*//p"
-  register: expected_matches
+  shell: grep "TOTAL EXPECTED MISSING MATCHES" "{{ test_out_dir }}/{{ summary_file }}" | sed -n "s/TOTAL EXPECTED MISSING MATCHES:[[:space:]]*//p"
+  register: expected_missing_matches
 
 - set_fact:
-    fail_in_logs: "{{ errors_found.stdout  != \"0\" or expected_matches.stdout != \"0\" }}" 
+    fail_in_logs: errors_found.stdout != "0" or expected_missing_matches.stdout != "0"
```
Comment (maggiemsft) [inline ansible/roles/test/files/tools/loganalyzer/loganalyzer_end.yml]: I think without the quote, it failed comparison while I tried it. Did you check this in your running? what's your ansible version?

Comment (marian-pritsak) [reply ansible/roles/test/files/tools/loganalyzer/loganalyzer_end.yml]: 2.0.0.2
I reverted this change.

**Pull request merged**